pkg/translator/loki/encode_test.go

```diff
@@ -83,50 +84,120 @@ func TestSerializeBody(t *testing.T) {
 	bytes.Bytes().FromRaw([]byte(`abc`))
 
 	testcases := []struct {
-		input    pcommon.Value
-		expected []byte
+		input          pcommon.Value
+		expectedJSON   []byte
+		expectedLogfmt []byte
 	}{
 		{
 			pcommon.NewValueEmpty(),
 			nil,
+			nil,
 		},
 		{
 			pcommon.NewValueStr("a"),
 			[]byte(`"a"`),
+			[]byte(`a=`),
 		},
 		{
 			pcommon.NewValueInt(1),
 			[]byte(`1`),
+			[]byte(`msg=1`),
 		},
 		{
 			pcommon.NewValueDouble(1.1),
 			[]byte(`1.1`),
+			[]byte(`msg=1.1`),
 		},
 		{
 			pcommon.NewValueBool(true),
 			[]byte(`true`),
+			[]byte(`msg=true`),
 		},
 		{
 			simplemap,
 			[]byte(`{"key":"val"}`),
+			[]byte(`key=val`),
 		},
 		{
 			complexmap,
 			[]byte(`{"keyarr":["a","b"],"keybool":true,"keyempty":null,"keyint":1,"keymap":{"key":"val"},"keynull":null,"keystr":"val"}`),
+			[]byte(`keystr=val keyint=1 keybool=true keyarr_0=a keyarr_1=b keymap_key=val`),
 		},
 		{
 			arrayval,
 			[]byte(`["a","b"]`),
+			[]byte(`body_0=a body_1=b`),
 		},
 		{
 			bytes,
 			[]byte(`"YWJj"`),
+			[]byte(`msg=abc`),
 		},
 	}
 
 	for _, test := range testcases {
-		out, err := serializeBody(test.input)
+		out, err := serializeBodyJSON(test.input)
+		assert.NoError(t, err)
```

Comment:
Also not for this PR, but "require" would be more appropriate here, as it's happening during the data preparation. The actual test target is the marshaling behavior (line 144), so, only the lines after that should use "assert".